### PR TITLE
Add resume

### PR DESCRIPTION
**Pull request by mhq199657**
This PR adds a **optional** field called `Resume`. The behaviour of this field is similar to the behaviour of the other existing optional field, `Tag`. Each `Person` will only have one `Resume`.

`Resume` will contain a value, which specifies the file name of the resume PDF. During add/edit command, the constructor will check whether the specified PDF file name is valid, in the sense that it should not be an empty string, the file should exist, the file size should be smaller than 1MB and the file should be a valid PDF file.

The `Resume` should be displayed after we complete the update of GUI.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Resume.java [hunk 1]
```diff
@@ -74,8 +78,18 @@ private static boolean isPdf(byte[] data) {
                     data[data.length - 5] == 0x45 && // E
```
Comment (tanhengyeow) [inline src/main/java/seedu/address/model/person/Resume.java]: Might want to link the reference to validation of pdf files :smiley:

Comment (Ang-YC) [reply src/main/java/seedu/address/model/person/Resume.java]: I believe he gets it from here: https://stackoverflow.com/questions/941813/how-can-i-determine-if-a-file-is-a-pdf-file

Besides, I might extract this part out to a utility class or use PDF library to validate when I update on UI during the next milestone

**Comment by Ang-YC** [comment]
It looks great, thanks for the hard work, I will try to do the UI in the upcoming milestone!

File: src/main/java/seedu/address/model/person/Resume.java (hunk 1 above)
Comment (mhq199657) [reply src/main/java/seedu/address/model/person/Resume.java]: I think Yee Chin's approach is better. My way of checking is quite stupid = =

**Pull request merged**